architecture.md

```diff
@@ -638,8 +652,36 @@ a pod is **not** `Privileged` the runtime selection is done as follows:
 
 ### Containerd CRI Plugin (Formerly CRI-containerd)
 
-The general guidelines for the Containerd CRI Plugin support is similar to the CRI-O support. You can run trusted workloads with a runtime like `runc` and then run an untrusted workload with Kata Containers. The parameters that you can modify in the containerd config to run Kata Containers along with another 'trusted' runtime are the following:
+The general guidelines for the Containerd CRI Plugin support is similar to the CRI-O support. You can run trusted workloads with a runtime like `runc` and then run an untrusted workload with Kata Containers. The parameters that you can modify in the containerd config to run Kata Containers along with another 'trusted' runtime are the following (with `shimv2` support):
+
+```
+# /etc/containerd/config.toml
+
+[plugins.cri]
+
+  [plugins.cri.containerd]
+  
+    # "plugins.cri.containerd.default_runtime" is the runtime to use in containerd.
+    [plugins.cri.containerd.default_runtime]
```

Comment:
containerd cri has moved to `plugins.cri.containerd.runtimes` configuration to better match kubernetes runtime class requirements. And `untrusted_workload_runtime ` is now [deprecated](https://github.com/containerd/cri/blob/master/docs/config.md). Please use the new format and thus something like:
```toml
    [plugins.cri.containerd]
      snapshotter = "overlayfs"
      no_pivot = false
    [plugins.cri.containerd.runtimes]
      [plugins.cri.containerd.runtimes.runc]
         runtime_type = "io.containerd.runc.v1"
         [plugins.cri.containerd.runtimes.runc.options]
           NoPivotRoot = false
           NoNewKeyring = false
           ShimCgroup = ""
           IoUid = 0
           IoGid = 0
           BinaryName = "runc"
           Root = ""
           CriuPath = ""
           SystemdCgroup = false
      [plugins.cri.containerd.runtimes.kata]
         runtime_type = "io.containerd.kata.v2"
         [plugins.cri.containerd.runtimes.kata.options]
           NoPivotRoot = false
           NoNewKeyring = false
           ShimCgroup = ""
           IoUid = 0
           IoGid = 0
           BinaryName = "/usr/local/bin/containerd-shim-kata-v2"
           Root = ""
           CriuPath = ""
           SystemdCgroup = false
      [plugins.cri.containerd.runtimes.katacli]
         runtime_type = "io.containerd.runc.v1"
         [plugins.cri.containerd.runtimes.katacli.options]
           NoPivotRoot = false
           NoNewKeyring = false
           ShimCgroup = ""
           IoUid = 0
           IoGid = 0
           BinaryName = "/usr/local/bin/kata-runtime"
           Root = ""
           CriuPath = ""
           SystemdCgroup = false
```
Then users can define new runtime class resource with one of these runtime handlers: `runc`, `kata` and `kata-cli`, and reference it in their pod config, as documented in https://kubernetes.io/docs/concepts/containers/runtime-class/

The example works for runc, kata-runtime and containerd-shim-kata-v2. Please suggest better names than `kata` and `katacli` being used there though. I listed them mostly to show that we can have multiple runtime handlers within one containerd instance.

Reply:
hmm, ok, will make an update for runtime class...